### PR TITLE
update godeps and use by makefile

### PR DESCRIPTION
**Pull request by Mashimiao**
update dependencies to the latest version
and add directory vendor as _workspace will be deprecated in go1.8

Signed-off-by: Ma Shimiao mashimiao.fnst@cn.fujitsu.com

### CODE REVIEW

**Comment by liangchenye** [comment]
I'm fine to update this.
But I prefer to do it after most other PRs been merged.

**Comment by mrunalp** [comment]
Yeah, let's get the other ones in.

**Comment by Mashimiao** [comment]
@opencontainers/runtime-tools-maintainers 
Can we let this in first? I think there are many differences between spec v1.0.0-rc1 and v1.0.0-rc2.
The backward spec-go has affected usage. like mentioned in #268

File: .travis.yml [hunk 1]
```diff
@@ -5,14 +5,15 @@ go:
 sudo: false
 
 before_install:
+  - go get github.com/tools/godep
```
Comment (wking) [inline .travis.yml]: Why this addition?  Does Travis start using `godep` somewhere?

Comment (Mashimiao) [inline .travis.yml]: godep will be used in Makefile

Comment (wking) [inline .travis.yml]: Ah, I'd missed that between all the `Godeps` removals and `vendor` additions.

**Comment by Mashimiao** [comment]
As #349 is landed, can we start to review this PR?
@liangchenye  @mrunalp

**Comment by Mashimiao** [comment]
ping @liangchenye  @mrunalp  @hqhq

**Comment by liangchenye** [comment]
`golint` in `make test` will check all the files under `vendor` directory, so the CI fails.

Can you skip this in Makefile? @Mashimiao

**Comment by Mashimiao** [comment]
@liangchenye  yes, updated. I think there is no need to exit with output of golint

**Comment by liangchenye** [comment]
LGTM

**Comment by hqhq** [comment]
@Mashimiao What tool are you using? Just godep or tools like vndr?

Why there's no config file for dependencies?

**Comment by Mashimiao** [comment]
Just use godep. sorry for leave the config behind.
@liangchenye  @hqhq  PTAL

File: Godeps/Godeps.json [hunk 2]
```diff
@@ -49,6 +50,10 @@
 			"ImportPath": "github.com/urfave/cli",
 			"Comment": "v1.19.1",
 			"Rev": "0bdeddeeb0f650497d603c4ad7b20cfe685682f6"
+		},
+		{
+			"ImportPath": "golang.org/x/sys/unix",
+			"Rev": "f3918c30c5c2cb527c0b071a27c35120a6c0719a"
```
Comment (hqhq) [inline Godeps/Godeps.json]: Why do you need to add this in this PR?

Comment (Mashimiao) [inline Godeps/Godeps.json]: automatically added by godep, not sure if it should be here. But as unix also is called and depended by runtime-tools, I think may be it's OK

Comment (Mashimiao) [inline Godeps/Godeps.json]: I find runc or image-tools also add them into vendor directory

Comment (hqhq) [inline Godeps/Godeps.json]: They use `x/sys/unix` to replace `syscall` package, maybe it's used by the updated vendors, it's fine if is automatically added.

File: Makefile [hunk 3]
```diff
@@ -31,25 +32,20 @@ uninstall:
...
 
 test: .gofmt .govet .golint
 
 .gofmt:
-	OUT=$$(go fmt ./...); if test -n "$${OUT}"; then echo "$${OUT}" && exit 1; fi
+	OUT=$$(go fmt ./... | grep -v vendor); if test -n "$${OUT}"; then echo "$${OUT}" && exit 1; fi
 
 .govet:
-	go vet -x ./...
+	OUT=$$(go vet ./... | grep -v "exit status" | grep -v vendor); if test -n "$${OUT}"; then echo "$${OUT}" && exit 1; fi
```
Comment (hqhq) [inline Makefile]: I would do something like:
```
packages = $(shell go list ./... | grep -v vendor)
.govet:
    go vet $(packages)
```

Comment (Mashimiao) [inline Makefile]: updated, PTAL

Comment (hqhq) [inline Makefile]: Not exactly what I expected, but we can do that later.

**Comment by hqhq** [comment]
LGTM

**Comment by liangchenye** [comment]
LGTM

**Pull request merged**